### PR TITLE
Remove --math-overflow flag

### LINKED ISSUE #1216: Always enable math overflow unless in unchecked {} block
The `--math-overflow` flag should be removed, and should always be enabled.

The `unchecked` fields in codegen should only be set for code in `unchecked {}` block or generated code which does not require overflow checks.

Most likely there are a lot of tests than need fixing.

### PR DESCRIPTION
**Pull request by seanyoung**
Fixes #1216

### CODE REVIEW

File: src/emit/expression.rs [hunk 1]
```diff
@@ -121,8 +121,8 @@ pub(super) fn expression<'a, T: TargetRuntime<'a> + ?Sized>(
             let left = expression(target, bin, l, vartab, function, ns).into_int_value();
             let right = expression(target, bin, r, vartab, function, ns).into_int_value();
 
-            if bin.options.math_overflow_check && !*unchecked {
-                let signed = l.ty().is_signed_int();
+            if !unchecked {
```
Comment (LucasSte) [inline src/emit/expression.rs:124]: Why aren't we dereferencing anymore?

File: src/emit/expression.rs [hunk 2]
```diff
@@ -143,8 +143,8 @@ pub(super) fn expression<'a, T: TargetRuntime<'a> + ?Sized>(
...
```
Comment (LucasSte) [inline src/emit/expression.rs:146]: Why aren't we dereferencing anymore?

**Review by LucasSte — APPROVED** [review]

**Review by xermicus — APPROVED** [review]
LGTM

File: src/emit/expression.rs (hunk 1 above)
Comment (seanyoung) [reply src/emit/expression.rs:124]: The deref is implicit

File: src/emit/expression.rs (hunk 2 above)
Comment (seanyoung) [reply src/emit/expression.rs:146]: The deref is implicit

**Pull request merged**